### PR TITLE
Handle methods with variadic parameters & fixed generated code for existential parameters

### PR DESCRIPTION
**Pull request by pocketal**
1 - handle methods with variadic parameters
2 - fixed generated code for existential parameters

### CODE REVIEW

**Comment by art-divin** [comment]
@rokridi Linux tests failed because in the modified stencil there is a call to isVariadic on type String and not a typoe from AST.

See the error text:
SourceryRuntime/TypeName_Linux.swift:31: Fatal error: unable to lookup: isVariadic in String?

File: Templates/Tests/Expected/AutoMockable.expected [hunk 1]
```diff
@@ -33,28 +33,27 @@ import AppKit
 
 
 
-public class AccessLevelProtocolMock: AccessLevelProtocol {
+class AccessLevelProtocolMock: AccessLevelProtocol {
 
-    public init() {}
 
-    public var company: String?
```
Comment (art-divin) [inline Templates/Tests/Expected/AutoMockable.expected:40]: Hey @rokridi , I wonder why public modifier is removed. It might be a regression

Comment (pocketal) [reply Templates/Tests/Expected/AutoMockable.expected]: fixed

**Comment by pocketal** [comment]
> @rokridi Linux tests failed because in the modified stencil there is a call to isVariadic on type String and not a type from AST.
> 
> See the error text: SourceryRuntime/TypeName_Linux.swift:31: Fatal error: unable to lookup: isVariadic in String?

How can I run tests locally ?

**Comment by art-divin** [comment]
> How can I run tests locally ?

I suggest to read Stencil carefully, and check all occurrences of `isVariadic` , at least one of them is used against String type.
It is faster and more reliable to do than if to setup a VM and dive into debugging in VSCode.

If you won't be able to deduct what's wrong, feel free to tell me, I might have a chance to check it myself later this week.

**Comment by art-divin** [comment]
Resolves #1267 resolves #1265

@rokridi to make GitHub understand what needs to be automatically closed, keyword "resolves" needs to be put in front of each issue number like here 👆🏻

Reference: https://github.blog/2013-05-14-closing-issues-via-pull-requests/

**Comment by art-divin** [comment]
👋🏻  @rokridi I have run tests for macOS variant, and here's my findings:

1. Extected file (AutoMockable.expected) is outdated in your branch. Please find the instructions how to update it [here](https://github.com/krzysztofzablocki/Sourcery/pull/1253).
2. Generated code for one of the mockable protocols is not valid swift:

```swift
class StaticMethodProtocolMock: StaticMethodProtocol {



    static func reset()
    {
         //MARK: - staticFunction
        staticFunctionStringStringCallsCount = 0
        staticFunctionStringStringReceived = nil
        staticFunctionStringStringReceivedInvocations = []
        staticFunctionStringStringClosure = nil


    }

    //MARK: - staticFunction

    static var staticFunctionStringStringCallsCount = 0
    static var staticFunctionStringStringCalled: Bool {
        return staticFunctionStringStringCallsCount > 0
    }
    static var staticFunctionStringStringReceived: (String)?
    static var staticFunctionStringStringReceivedInvocations: [(String)] = []
    static var staticFunctionStringStringReturnValue: String!
    static var staticFunctionStringStringClosure: ((String) -> String)?

    static func staticFunction(_ : String) -> String {
        staticFunctionStringStringCallsCount += 1
        staticFunctionStringStringReceived =
        staticFunctionStringStringReceivedInvocations.append()
        if let staticFunctionStringStringClosure = staticFunctionStringStringClosure {
            return staticFunctionStringStringClosure()
        } else {
            return staticFunctionStringStringReturnValue
        }
    }


}
```

Everything else seems to be OK within the macOS tests context! Great work! 
🤝

**Comment by art-divin** [comment]
👋🏻  Hey @pocketal 

Do you need any help with this PR?
I could take over if you want, I am almost done with 2.1.8 release preparation at the moment, and I would love to include this fix into it.

Thanks for your contribution 👍🏻 

---

_If I would take over, I would cherry-pick your commits into a new PR and continue from there, persisting attribution._

**Comment by AliouSARR** [comment]
Hey @art-divin 
Sorry I was very busy these days and didn't have time to look into it.
You can take over if you want.
Thanks for you help 🙏

**Comment by art-divin** [comment]
Closing in favor of #1304

**Pull request closed without merging**